Jenkins integration

drothlis opened the pull request:
Extend the command-line API so that it is convenient for using from Jenkins; and polish the help & documentation to a releasable state.

For now we've descoped the Python API completely; the only public API is the command-line interface.

TODO:

- [x] Flesh out command-line API.
- [x] Implement it.
- [x] Test interactive mode.
- [x] Test jenkins mode.
- [ ] Write instructions for Jenkins job configuration.

drothlis committed:
Flesh out command-line API

Not all the arguments are implemented as documented (or at all) -- that
will come in future commits.

drothlis committed:
Don't try invalid authentication token twice

drothlis committed:
Fix exit status on authentication failure

drothlis committed:
Handle all the command-line arguments for "run"

drothlis committed:
Remove duplicate "stbt-rig" from log message

drothlis committed:
Remove duplicate error message

drothlis committed:
Use github username for default category & branch name

drothlis committed:
Print results in interactive mode, save jUnit xml in jenkins mode

drothlis committed:
Record Jenkins variables (BUILD_ID etc) as tags

drothlis committed:
Fix TypeError: TestJob.Status() takes no parameters

Introduced in db507eb (Remove dependency on enum34).

I've removed class `TestJob.Status` to make it clear that it isn't an
enum.

drothlis committed:
Refactor: Use logger levels instead of passing verbosity to TestPack

drothlis committed:
Add feedback at INFO level

Otherwise it just sits there until all the tests have finished and you
don't have any indication that the tests have started. In future we
might want to stream the job log.

wmanley commented on stbt_rig.py:
```diff
@@ -1,89 +1,197 @@
+              See https://stb-tester.com/manual/rest-api-v2#authentication
+
+            INTERACTIVE MODE:
+              blah blah
```
Hoping this documentation will be improved in future commits ;)

Hmm, maybe not?

wmanley commented on stbt_rig.py:
```diff
@@ -1,89 +1,197 @@
+            Note: Run "./stbt_rig.py COMMAND -h" to see the additional
+            parameters for each COMMAND."""))
+
+    run_parser = subcommands.add_parser("run", help="Run test-cases")
```
Perhaps we should refer to the `run_tests` REST API documentation here as a useful cross-reference.

drothlis committed:
Print link to test-results view at end of test run

drothlis committed:
Don't show stacktrace for NodeBusyException

drothlis committed:
Stop TestJob on SIGTERM or SIGHUP

By handling the exception and calling sys.exit, the Python `finally`
blocks or context manager `__exit__` methods will be run.

By handling SIGINT this way too, we avoid a big stack trace when you
press Control-C.

drothlis committed:
Finish the documentation on authentication, interactive mode, and jenkins
